pylint/utils.py

```diff
@@ -730,32 +732,93 @@ def register_messages(self, checker):
+        self._alternative_names[msgid] = msg
+        self._alternative_names[symbol] = msg
+
+    def _check_symbol(self, msgid, symbol):
+        """ Check that a symbol is not already used. """
+        if symbol in self._messages:
+            other_msgid = self._messages[symbol].msgid
+            self._raise_ime_duplicate_msgid(symbol, msgid, other_msgid)
+        if symbol in self._alternative_names:
+            msg = self._alternative_names[symbol]
+            if msg.symbol == symbol:
+                other_msgid = msg.msgid
+            else:
+                for old_msgid, old_symbol in msg.old_names:
+                    if old_symbol == symbol:
+                        other_msgid = old_msgid
+                        break
+            if msgid != other_msgid:
+                self._raise_ime_duplicate_msgid(symbol, msgid, other_msgid)
+
+    def _check_msgid(self, msgid, symbol):
+        for m_symbol in self._messages:
+            msg = self._messages[m_symbol]
+            if msg.msgid == msgid:
+                self._raise_ime_duplicate_symbol(msgid, symbol, msg.symbol)
+
+    def _check_message(self, msg, chkid=None, is_old_name=False):
+        """A new messages is correct according to a dict of messages. """
+        msgid, symbol, old_names = msg.msgid, msg.symbol, msg.old_names
+        self._check_id_and_symbol_consistency(msgid, symbol)
+        self._check_symbol(msgid, symbol)
+        self._check_msgid(msgid, symbol)
+        for old_name in old_names:
+            self._check_symbol(msgid, old_name[1])
+        if chkid is not None and chkid != msgid[1:3]:
+            ime = "Inconsistent checker part in message id %r" % msgid
+            ime += " (expected 'x%sxx')" % chkid
+            raise InvalidMessageError(ime)
+
+    def _check_id_and_symbol_consistency(self, msgid, symbol):
+        try:
+            alternative = self._alternative_names[msgid]
+        except KeyError:
+            alternative = False
+        try:
+            if not alternative:
+                alternative = self._alternative_names[symbol]
+        except KeyError:
+            # There is no alternative names concerning this msgid/symbol.
+            # So nothing to check
+            return None
+        old_symbolic_name = None
+        old_symbolic_id = None
+        for tuple_ in alternative.old_names:
+            if tuple_[0] == msgid or tuple_[1] == symbol:
+                old_symbolic_id = tuple_[0]
+                old_symbolic_name = tuple_[1]
+        if alternative.symbol != symbol and symbol != old_symbolic_name:
+            if msgid == old_symbolic_id:
+                self._raise_ime_duplicate_symbol(msgid, symbol,
+                                                 old_symbolic_name)
+            else:
+                self._raise_ime_duplicate_symbol(msgid, symbol,
+                                                 alternative.symbol)
+
+    def _raise_ime_duplicate_symbol(self, msgid, symbol, other_symbol):
```

Comment:
I am being picky, but it isn't clear what `ime` is until you read the docstring. Can we make this method name more clear? Perhaps simply `_raise_duplicate_symbol` and `_raise_duplicate_msg_id`?